### PR TITLE
[NFC] Clang-format DXC sources

### LINKED ISSUE #5591: clang-format DXC sources
**Description**
DXC sources are inconsistent in their style and formatting. We should clang-format the codebase to adopt consistent style.

This file lists all the files unique to DXC. We should clang-format all these files to adopt consistent style.

[dxc-source-files.txt](https://github.com/microsoft/DirectXShaderCompiler/files/12422000/dxc-source-files.txt)

**Environment**
- DXC version: `main`
- Host Operating System: all

### PR DESCRIPTION
**Pull request by llvm-beanz**
This applies clang-format with the LLVM style to the DXC codebase.

Fixes #5591

### CODE REVIEW

File: tools/clang/lib/Sema/SemaHLSL.cpp [hunk 1]
```diff
@@ -852,72 +880,45 @@ QualType GetOrCreateVectorSpecialization(ASTContext& context, Sema* sema,
...
+// replaced with clang::Type-based constructs.
 
 static const LPCSTR kBuiltinIntrinsicTableName = "op";
 
-static const ArTypeObjectKind g_ScalarTT[] =
-{
-  AR_TOBJ_SCALAR,
-  AR_TOBJ_UNKNOWN
-};
+static const ArTypeObjectKind g_ScalarTT[] = {AR_TOBJ_SCALAR, AR_TOBJ_UNKNOWN};
```
Comment (pow2clk) [inline tools/clang/lib/Sema/SemaHLSL.cpp:906]: I feel like this is a loss of readability. If we had comments after each entry, would clang-format let us keep the multiple lines?

Comment (llvm-beanz) [reply tools/clang/lib/Sema/SemaHLSL.cpp:906]: Yes, clang will keep it if there are trailing comments. I don't want to go through and make a bunch of hand edits here to make people happy about individual formatting changes.

This change is about adopting a policy and enforcing it.

File: tools/clang/unittests/HLSL/VerifierTest.cpp [hunk 2]
```diff
@@ -122,17 +123,19 @@ class VerifierTest TEST_CLASS_DERIVATION {
...
+    char *commandLine;
 
     //
     // Very simple processing for now.
     // See utils\lit\lit\TestRunner.py for the more thorough implementation.
     //
     // The first line for HLSL tests will always look like this:
-    // // RUN: %clang_cc1 -fsyntax-only -Wno-unused-value -ffreestanding -verify %s
+    // // RUN: %clang_cc1 -fsyntax-only -Wno-unused-value -ffreestanding -verify
+    // %s
```
Comment (bogner) [inline tools/clang/unittests/HLSL/VerifierTest.cpp:134]: This scared me for a second, but it's just a comment

File: include/dxc/Support/Unicode.h [hunk 3]
```diff
@@ -33,65 +33,80 @@ int WideCharToMultiByte(uint32_t CodePage, uint32_t dwFlags,
...
 // A ccp_char is a character encoded in the console code page.
 typedef char ccp_char;
 
-_Success_(return != false)
-bool UTF8ToConsoleString(_In_opt_count_(textLen) const char* text, _In_ size_t textLen, _Inout_ std::string* pValue, _Out_opt_ bool* lossy);
+_Success_(return != false) bool UTF8ToConsoleString(_In_opt_count_(textLen)
+                                                        const char *text,
+                                                    _In_ size_t textLen,
+                                                    _Inout_ std::string *pValue,
+                                                    _Out_opt_ bool *lossy);
```
[the remark this answers is not in the document]
Comment (bogner) [reply include/dxc/Support/Unicode.h:60]: I think this is the only one of my comments that's particularly important - the other ones look kind of bad but we can clean them up after the fact as people run into them.

**Review by bogner — APPROVED** [review]

File: include/dxc/Support/Unicode.h (hunk 3 above)
[the remark this answers is not in the document]
Comment (llvm-beanz) [reply include/dxc/Support/Unicode.h:60]: @pow2clk, @tex3d, is there any reason not to just remove all the SAL annotations? They trip up clang format, but we also use them inconsistently enough that I wonder if they bring value.

I can maybe see keeping them on public APIs, but for internal APIs it seems like they don't add much value.

**Review by s-perron — APPROVED** [review]

File: tools/clang/lib/Sema/SemaHLSL.cpp (hunk 1 above)
Comment (pow2clk) [reply tools/clang/lib/Sema/SemaHLSL.cpp:906]: You don't want to do it, but would you object if I did?

Comment (llvm-beanz) [reply tools/clang/lib/Sema/SemaHLSL.cpp:906]: I have no objection to fixing formatting issues. You can do it before I merge the clang-format change or after. Either way is fine.

**Pull request merged**